### PR TITLE
Added privacy manifest

### PR DESCRIPTION
**Pull request by oldalton**
## Proposed changes

Per new requirements from Apple:
https://developer.apple.com/documentation/bundleresources/privacy_manifest_files/describing_data_use_in_privacy_manifests?language=objc
https://developer.apple.com/documentation/bundleresources/privacy_manifest_files?language=objc#4284009

## Type of change

- [ ] Feature work
- [ ] Bug fix
- [x] Documentation
- [ ] Engineering change
- [ ] Test
- [ ] Logging/Telemetry

## Risk

- [ ] High – Errors could cause MAJOR regression of many scenarios. (Example: new large features or high level infrastructure changes)
- [ ] Medium – Errors could cause regression of 1 or more scenarios. (Example: somewhat complex bug fixes, small new features)
- [x] Small – No issues are expected. (Example: Very small bug fixes, string changes, or configuration settings changes)

### CODE REVIEW

**Review by juan-arias — APPROVED** [review]

**Review by Veena11 — APPROVED** [review]

File: MSAL/PrivacyInfo.xcprivacy [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+			<key>NSPrivacyCollectedDataTypeTracking</key>
+			<false/>
+			<key>NSPrivacyCollectedDataTypePurposes</key>
+			<array>
+				<string>NSPrivacyCollectedDataTypePurposeAppFunctionality</string>
+			</array>
+		</dict>
+		<dict>
+			<key>NSPrivacyCollectedDataType</key>
+			<string>NSPrivacyCollectedDataTypeOtherUserContactInfo</string>
```
Comment (Veena11) [inline MSAL/PrivacyInfo.xcprivacy:45]: What comes under UserContactInfo other than email address and user id ?. Is it phone number for MFA ?

File: MSAL/PrivacyInfo.xcprivacy [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+			<key>NSPrivacyAccessedAPITypeReasons</key>
+			<array>
+				<string>1C8F.1</string>
+				<string>CA92.1</string>
+			</array>
+			<key>NSPrivacyAccessedAPIType</key>
+			<string>NSPrivacyAccessedAPICategoryUserDefaults</string>
+		</dict>
+	</array>
+	<key>NSPrivacyCollectedDataTypes</key>
```
Comment (nilo-ms) [inline MSAL/PrivacyInfo.xcprivacy:17]: I think we didn't miss anything here, but @spetrescu84 could you please double check?

File: MSAL/PrivacyInfo.xcprivacy [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+	<array>
```
Comment (nilo-ms) [inline MSAL/PrivacyInfo.xcprivacy:18]: Should we mention also "NSPrivacyCollectedDataTypePhoneNumber" ?
More details here: https://developer.apple.com/documentation/bundleresources/privacy_manifest_files/describing_data_use_in_privacy_manifests#4250555

Comment (oldalton) [reply MSAL/PrivacyInfo.xcprivacy:18]: Do you collect phone number in CIAM APIs?

File: MSAL/PrivacyInfo.xcprivacy (hunk 2 above)
Comment (oldalton) [reply MSAL/PrivacyInfo.xcprivacy:17]: @nilo-ms, @spetrescu84, can you please check before end of this week?

File: MSAL/PrivacyInfo.xcprivacy (hunk 1 above)
Comment (oldalton) [reply MSAL/PrivacyInfo.xcprivacy:45]: Anything that Entra might be collecting as part of account setup, e.g. job title, job location, etc

File: MSAL/PrivacyInfo.xcprivacy (hunk 3 above)
Comment (nilo-ms) [reply MSAL/PrivacyInfo.xcprivacy:18]: On native auth APIs we don't accept yet phone number as username. So, we don't collect it. But on web, if I'm not wrong, a phone number can be used as username. What happen in this case? Is the username (phone number) collected from the SDK? If not, you can resolve this conversation

File: MSAL/PrivacyInfo.xcprivacy (hunk 2 above)
Comment (spetrescu84) [reply MSAL/PrivacyInfo.xcprivacy:17]: The only one I see is maybe `NSPrivacyCollectedDataTypeOtherDataTypes` for the Telemetry? We collect API results and so on in the `x-client-last-telemetry` and `x-client-current-telemetry` headers?
Or all of that falls under `NSPrivacyCollectedDataTypeOtherDiagnosticData` ?

Comment (spetrescu84) [reply MSAL/PrivacyInfo.xcprivacy:17]: To add to this, we haven't added anything in Native Auth that collects any other data that's not already collected in MSAL Objective C. We do have new endpoints and such in the telemetry for example, but don't have new fields or own telemetry

Comment (oldalton) [reply MSAL/PrivacyInfo.xcprivacy:17]: Ok, sounds good

**Pull request merged**